hazelcast/src/main/java/com/hazelcast/map/impl/proxy/MapProxySupport.java

```diff
@@ -1288,13 +1288,15 @@ public void executeOnEntriesInternal(EntryProcessor entryProcessor, Predicate pr
             Map<Integer, Object> results;
             if (predicate instanceof PartitionPredicate) {
                 PartitionPredicate partitionPredicate = (PartitionPredicate) predicate;
-                Data key = toData(partitionPredicate.getPartitionKey());
```

Comment:
Regarding my comment above, I don't understand why we were not using `toDataWithStrategy` to serialize the partition key in the old implementation here and in some other places. Maybe a bug in our end or using the `toDataWithStrategy` above `removeAllInternal` method was a mistake? I feel like the former. WDYT @ahmetmircik

Reply:
yeah, this seems a bug.